packages/slice-machine/src/domain/customType.ts

```diff
@@ -425,6 +447,41 @@ export function reorderField(args: ReorderFieldArgs): CustomType {
   return newCustomType;
 }
 
+// TODO add tests
+export function updateUidField(
+  label: string,
+  customType: CustomType,
+): CustomType {
+  const field = getUidField(customType);
+  if (!field) return customType;
+
+  const section = findSectionWithUidField(customType);
+  const sectionId = section ?? "Main";
```

Comment:
⚠️ #issue

2 things for me.

1 - I don't understand the management, the function is name `updateUidField`, and if you don't find a section with a field in it, you still want to update it, but then it's creating it. Is that what we want? If yes maybe the name should be clearer.

2 - We cannot rely on "Main" directly for me here. I would say that you want to create also and that's why you added the "Main" If you want to "add" you need to get the main section id, because it can NOT be "Main", it's technically possible. Then with this code you would create on the fly another section for the user.

Reply:
need to re-think that, but clearly it needs some fixes, basicly the uid is already attached to main section so we shouldn't need to create the field or the section